packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -218,6 +235,156 @@ export function getComponentTracker(
     return updatedComponentTracker;
 }
 
+/**
+ * Creates a schema location from a data location
+ */
+export function mapSchemaLocationFromDataLocation(dataLocation: string, data: any, schema: any): string {
+    const squareBracketRegex: RegExp = getRegex(RegexType.squareBracket);
+    let normalizedDataLocation: string = dataLocation.replace(squareBracketRegex, `.$1`); // convert all [ ] to . notation
+    normalizedDataLocation = convertArrayItemsToBracketNotation(dataLocation, data); // convert back all array items to use [ ]
+    const dataLocationSegments: string[] = normalizedDataLocation.split(".");
+
+    if (dataLocation === "") {
```

Comment:
It seems like you should perform this check prior to any other work because you know as soon as the function executes if you should exit